src/conn/conn_sweep.c

```diff
@@ -359,8 +366,14 @@ __wt_sweep_create(WT_SESSION_IMPL *session)
 	/*
 	 * Handle sweep does enough I/O it may be called upon to perform slow
 	 * operations for the block manager.
+	 *
+	 * The sweep thread sweeps the lookaside table for outdated records,
+	 * it gets its own cursor for that purpose.
+	 *
+	 * Don't tap the sweep thread for eviction.
 	 */
-	F_SET(session, WT_SESSION_CAN_WAIT);
+	F_SET(session, WT_SESSION_CAN_WAIT |
+	    WT_SESSION_LOOKASIDE_CURSOR | WT_SESSION_NO_EVICTION);
```

Comment:
I don't think that's better, "LRU" eviction happens at the top level too.

How about `WT_SESSION_NO_WAIT_EVICTION`, or `WT_SESSION_NO_OPPORTUNISTIC_EVICTION`?

The latter is awfully long, but maybe something explicit that grabs attention is right thing to do here.